Prompt download for models in captive portal mode

pascoej opened the pull request:
#### a34fa2d34bd998ddd6fe2f24d1260632abea3340
<pre>
Prompt download for models in captive portal mode
<a href="https://bugs.webkit.org/show_bug.cgi?id=243061">https://bugs.webkit.org/show_bug.cgi?id=243061</a>
rdar://97381599

Reviewed by Dean Jackson.

* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::decidePolicyForResponseShared):
We should prompt for download when viewing model files instead of showing them.
* Tools/TestWebKitAPI/Tests/WebKitCocoa/Download.mm:
(tempUSDZThatDoesNotExist):
Add test for download prompt.

Canonical link: <a href="https://commits.webkit.org/252798@main">https://commits.webkit.org/252798@main</a>
</pre>

cdumez commented on Source/WebKit/UIProcess/WebPageProxy.cpp:
```diff
@@ -5948,7 +5948,7 @@ void WebPageProxy::decidePolicyForResponseShared(Ref<WebProcessProxy>&& process,
         });
 #if USE(QUICK_LOOK)
-        if (process->captivePortalMode() == WebProcessProxy::CaptivePortalMode::Enabled && (MIMETypeRegistry::isPDFOrPostScriptMIMEType(navigationResponse->response().mimeType()) || PreviewConverter::supportsMIMEType(navigationResponse->response().mimeType())))
+        if (process->captivePortalMode() == WebProcessProxy::CaptivePortalMode::Enabled && (MIMETypeRegistry::isPDFOrPostScriptMIMEType(navigationResponse->response().mimeType()) || MIMETypeRegistry::isSupportedModelMIMEType(navigationResponse->response().mimeType()) || PreviewConverter::supportsMIMEType(navigationResponse->response().mimeType())))
```
Not new to this patch but I think this is slightly wrong. What if the client app responded with PolicyAction::Ignore ? Then we are going to download the file instead of ignoring it like the client asked...

I think we should only override to `PolicyAction ::Download` if `policyAction` is `PolicyAction::Use`.

cdumez commented on Source/WebKit/UIProcess/Cocoa/WKWebViewContentProviderRegistry.mm:
```diff
@@ -56,7 +57,7 @@ - (instancetype)initWithConfiguration:(WKWebViewConfiguration *)configuration
 
 #if USE(SYSTEM_PREVIEW) && !HAVE(UIKIT_WEBKIT_INTERNALS)
-    if (configuration._systemPreviewEnabled && !configuration.preferences._modelDocumentEnabled) {
+    if (![_WKSystemPreferences isCaptivePortalModeEnabled] && configuration._systemPreviewEnabled && !configuration.preferences._modelDocumentEnabled) {
```
This looks suspicious. Even though lock down mode may be active at global/OS level, it may be disabled for the current site / view. Therefore, making decisions purely on the global flag is likely wrong

pascoej replied:
We don't do checks in the UI process for other disabled content types, so just removing this for now.

pascoej committed:
Prompt download for models in captive portal mode
https://bugs.webkit.org/show_bug.cgi?id=243061
rdar://97381599

Reviewed by Dean Jackson.

* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::decidePolicyForResponseShared):
We should prompt for download when viewing model files instead of showing them.
* Tools/TestWebKitAPI/Tests/WebKitCocoa/Download.mm:
(tempUSDZThatDoesNotExist):
Add test for download prompt.

Canonical link: https://commits.webkit.org/252798@main
